azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -1014,181 +1033,12 @@ func flattenKubernetesClusterAccessProfile(profile containerservice.ManagedClust
-		enabled := false
-		if enabledVal := httpApplicationRouting.Enabled; enabledVal != nil {
-			enabled = *enabledVal
-		}
-
-		zoneName := ""
-		if v := httpApplicationRouting.Config["HTTPApplicationRoutingZoneName"]; v != nil {
-			zoneName = *v
-		}
-
-		output := map[string]interface{}{
-			"enabled":                            enabled,
-			"http_application_routing_zone_name": zoneName,
-		}
-		routes = append(routes, output)
-	}
-	values["http_application_routing"] = routes
-
-	agents := make([]interface{}, 0)
-	if omsAgent := profile["omsagent"]; omsAgent != nil {
-		enabled := false
-		if enabledVal := omsAgent.Enabled; enabledVal != nil {
-			enabled = *enabledVal
-		}
-
-		workspaceId := ""
-		if workspaceResourceID := omsAgent.Config["logAnalyticsWorkspaceResourceID"]; workspaceResourceID != nil {
-			workspaceId = *workspaceResourceID
-		}
-
-		output := map[string]interface{}{
-			"enabled":                    enabled,
-			"log_analytics_workspace_id": workspaceId,
-		}
-		agents = append(agents, output)
-	}
-	values["oms_agent"] = agents
-
-	aciConnectors := make([]interface{}, 0)
-	if aciConnector := profile["aciConnectorLinux"]; aciConnector != nil {
-		enabled := false
-		if enabledVal := aciConnector.Enabled; enabledVal != nil {
-			enabled = *enabledVal
-		}
-
-		subnetName := ""
-		if v := aciConnector.Config["SubnetName"]; v != nil {
-			subnetName = *v
-		}
-
-		output := map[string]interface{}{
-			"enabled":     enabled,
-			"subnet_name": subnetName,
-		}
-		aciConnectors = append(aciConnectors, output)
-	}
-	values["aci_connector_linux"] = aciConnectors
-
-	kubeDashboards := make([]interface{}, 0)
-	if kubeDashboard := profile["kubeDashboard"]; kubeDashboard != nil {
-		enabled := false
-		if enabledVal := kubeDashboard.Enabled; enabledVal != nil {
-			enabled = *enabledVal
-		}
-
-		output := map[string]interface{}{
-			"enabled": enabled,
-		}
-		kubeDashboards = append(kubeDashboards, output)
-	}
-	values["kube_dashboard"] = kubeDashboards
-
-	azurePolicies := make([]interface{}, 0)
-	if azurePolicy := profile["azurepolicy"]; azurePolicy != nil {
-		enabled := false
-		if enabledVal := azurePolicy.Enabled; enabledVal != nil {
-			enabled = *enabledVal
-		}
-
-		output := map[string]interface{}{
-			"enabled": enabled,
-		}
-		azurePolicies = append(azurePolicies, output)
-	}
-	values["azure_policy"] = azurePolicies
-
-	return []interface{}{values}
-}
-
-func expandKubernetesClusterAgentPoolProfiles(d *schema.ResourceData) ([]containerservice.ManagedClusterAgentPoolProfile, error) {
-	configs := d.Get("agent_pool_profile").([]interface{})
-
+func expandKubernetesClusterAgentPoolProfiles(input []interface{}, isNewResource bool) ([]containerservice.ManagedClusterAgentPoolProfile, error) {
 	profiles := make([]containerservice.ManagedClusterAgentPoolProfile, 0)
-	for config_id := range configs {
-		config := configs[config_id].(map[string]interface{})
+
+	// TODO: fix this
```

Comment:
Is this still something that needs fixed?

Reply:
needs fixing - will push